examples/bzlmod/musl_cross_compiling/README.md

```diff
@@ -0,0 +1,779 @@
+
+This example code showcases how to cross-compile a simple Tokio API as a statically linked MUSL binary and how to package this binary in a scratch container. Furthermore, the example code eliminates a non-trivial performance bottleneck in MUSL via a custom memory allocator. 
+
+To build the example code:
+
+`bazel build //...`
+
+This may take a while on the first run because Bazel downloads multiple toolchains.
+
+To apply compiler optimization and striping the binary:
+
+`bazel build -c opt //...`
+
+The container supports both, Intel and ARM, architectures via a multi-arch OCI container image. To build the image:
+
+`
+bazel build //musl_scratch:image_index
+`
+
+To publish the container, you would have to run:
+
+`bazel run //musl_scratch:push`
+
+However, this results in an error because no registry is defined in the example.
+You have to define your target registry in the [binary BUILD](musl_scratch/BUILD.bazel) file
+before you can publish your image. For details of how to configure a container registry,
+please [consult the official documentation.](https://github.com/bazel-contrib/rules_oci/blob/main/docs/push.md)
+
+
+## MUSL specifics
+
+When you develop applications in Rust that target the MUSL, there are few important considerations:
+
+### 1) Replace SSL dependencies with RustTLS
+
+Because OpenSSL depends on some system libraries, any crate that depends on the default SSL will not compile with MUSL. Luckily, all major crates offer a RustTLS feature flag, so this is resolved by setting 
+the correct feature flags in the Rust dependencies. In general, it is easier to disable all default features, and then only enable the features that are actually needed. See the dependencies in the MODULE.bazel file as an example.
+
+### 2) Avoid major C/C++ dependencies whenever possible
+
+This is a non-trivial constraint, but not every C/C++ target compiles with MUSL. Unfortunately, there is no
+other way to find out than actually trying to build it. As a rule of thumb, it is best to avoid Rust crates that just wrap an existing C/C++ whenever possible. Many Rust crates have some implicit C/C++ dependencies, but many of these actually build with MUSL. However, the larger the underlying C/C++ library, the higher the chance that something breaks. 
+
+The exact problem isn't C/C++ per se, it's the problem that, just one call in std libc that tries to open a file or something else that's not supported in MUSL binaries simply breaks the build. This is equally true for Rust crates, but many Rust crates use feature flags to mitigate IO related issues. Some code out there actually requires system IO, and that just doesn't compile with MUSL. If somehow one or more major library is needed and cannot be build with MUSL, it's sensible to just move on, a build a dynamically linked binary and use a distroless image instead, as shown in the [OCI example](../07-oci-container).
+
+### 3) MUSL builds best with network only applications
```

Comment:
Is a "network only application" a common term? I've not come across it before, and am not entirely sure what it's referring to here...

Reply:
I just made the term up while writing this. it just means, your service is using only network to load whatever it needs. I am writing these kinds of micro services for years and build them static binaries w/o (g)libc due to security requirements. In GoLang, that's largely a non-event, but in Rust with MUSL, well, it's still a learning process for me. 

However, I agree that's not the best wording so let me edit this

Reply:
As this example gets removed for this PR, and most likely renovated once the MUSL issue is fixed,.
I have to re-write this Readme anyways, but  I put in a note for this.